devicetypes/smartthings/zwave-metering-switch-secure.src/zwave-metering-switch-secure.groovy

```diff
@@ -0,0 +1,320 @@
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "manufacturerId:   ${cmd.manufacturerId}"
+	log.debug "manufacturerName: ${cmd.manufacturerName}"
+	log.debug "productId:        ${cmd.productId}"
+	log.debug "productTypeId:    ${cmd.productTypeId}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.DeviceSpecificReport cmd) {
+	log.debug "deviceIdData:                ${cmd.deviceIdData}"
+	log.debug "deviceIdDataFormat:          ${cmd.deviceIdDataFormat}"
+	log.debug "deviceIdDataLengthIndicator: ${cmd.deviceIdDataLengthIndicator}"
+	log.debug "deviceIdType:                ${cmd.deviceIdType}"
+
+	if (cmd.deviceIdType == 1 && cmd.deviceIdDataFormat == 1) {//serial number in binary format
+		String serialNumber = "h'"
+
+		cmd.deviceIdData.each{ data ->
+			serialNumber += "${String.format("%02X", data)}"
+		}
+
+		updateDataValue("serialNumber", serialNumber)
+		log.debug "${device.displayName} - serial number: ${serialNumber}"
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	updateDataValue("version", "${cmd.applicationVersion}.${cmd.applicationSubVersion}")
+	log.debug "applicationVersion:      ${cmd.applicationVersion}"
+	log.debug "applicationSubVersion:   ${cmd.applicationSubVersion}"
+	log.debug "zWaveLibraryType:        ${cmd.zWaveLibraryType}"
+	log.debug "zWaveProtocolVersion:    ${cmd.zWaveProtocolVersion}"
+	log.debug "zWaveProtocolSubVersion: ${cmd.zWaveProtocolSubVersion}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [ displayed: true ]
+	if (cmd.sensorType == 4) {
+		createEvent(name: "power", value: Math.round(cmd.scaledSensorValue), unit: "W")
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	createEvent(name: "switch", value: cmd.value ? "on" : "off", type: "digital")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.deviceresetlocallyv1.DeviceResetLocallyNotification cmd) {
+	log.info "${device.displayName}: received command: $cmd - device has reset itself"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd)
+{
+	if (cmd.notificationType == 0x08) {
+		if (cmd.event == 0x06) {
+			createEvent(descriptionText: "Warning: $device.displayName detected over-current", isStateChange: true)
+		} else if (cmd.event == 0x08) {
+			createEvent(descriptionText: "Warning: $device.displayName detected over-load", isStateChange: true)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.debug "$device.displayName: Unhandled: $cmd"
+	[:]
+}
+
+// handle commands
+def configure() {
+	log.debug "Executing 'configure'"
+
+	def cmds = []
+
+	cmds += zwave.manufacturerSpecificV1.manufacturerSpecificGet()
+	cmds += zwave.manufacturerSpecificV2.deviceSpecificGet()
+	cmds += zwave.versionV1.versionGet()
```

Comment:
Please remove manufacturerSpecificGet and versionGet, we now get those and they are accessible in zwInfo.